mcp/gcp_cloudrun.tf

```diff
@@ -0,0 +1,162 @@
+      annotations = length(each.value.metadata) > 0 ? lookup(each.value.metadata, "annotations", null) : null
+    }
+  }
+  dynamic "traffic" {
+    for_each = local.cloudrun_traffic[each.key]
+    //noinspection HILUnresolvedReference
+    content{
+      percent         = lookup(traffic.value, "percent", null)
+      revision_name   = lookup(traffic.value, "revision_name", null)
+      latest_revision = lookup(traffic.value, "latest_revision", null)
+    }
+  }
+
+}
+
+data "google_iam_policy" "noauth" {
+  binding {
+    role    = "roles/run.invoker"
+    members = ["allUsers"]
+  }
+}
+
+data "google_iam_policy" "auth" {
+  //noinspection HILUnresolvedReference
+  for_each = local.cloudrun_specs
+  binding {
+    role    = "roles/${lookup(local.cloudrun_iam[each.key], "role", "" )}"
+    members = local.cloudrun_iam_members[each.key]
+  }
+}
+
+//noinspection HILUnresolvedReference
+resource "google_cloud_run_service_iam_policy" "policy" {
+  for_each    = {
+    for key, specs in local.cloudrun_specs: key => specs
+      if lookup(local.cloudrun_iam[key], "replace_policy", true)
+  }
+  location    = google_cloud_run_service.self[each.key].location
+  project     = google_cloud_run_service.self[each.key].project
+  service     = google_cloud_run_service.self[each.key].name
+
+  policy_data = each.value.auth ? data.google_iam_policy.auth[each.key].policy_data : data.google_iam_policy.noauth.policy_data
+}
+
+//noinspection HILUnresolvedReference
+resource "google_cloud_run_service_iam_binding" "binding" {
+  for_each = {
+    for key, specs in local.cloudrun_specs: key => specs
+      if local.cloudrun_specs[key].auth && lookup(local.cloudrun_iam[key], "binding", false)
+  }
+  project  = google_cloud_run_service.self[each.key].project
+  location = google_cloud_run_service.self[each.key].location
+  members  = local.cloudrun_iam_members[each.key]
+  role     = "roles/${lookup(local.cloudrun_iam[each.key], "role", "" )}"
+
+  service  = google_cloud_run_service.self[each.key].name
+}
+
+//noinspection HILUnresolvedReference
+resource "google_cloud_run_service_iam_member" "member" {
+  for_each = {
+    for key, specs in local.cloudrun_specs: key => specs
+      if local.cloudrun_specs[key].auth && lookup(local.cloudrun_iam[key], "add_member", false) == true
+  }
+  project  = google_cloud_run_service.self[each.key].project
+  location = google_cloud_run_service.self[each.key].location
+  member   = lookup(local.cloudrun_iam[each.key],"add_member", null ) == null ? "" : "${local.cloudrun_iam[each.key].add_member.member_type}:${local.cloudrun_iam[each.key].add_member.member}"
+  role     = lookup(local.cloudrun_iam[each.key],"add_member", null ) == null ? "" : "roles/${local.cloudrun_iam[each.key].add_member.role}"
+
+  service = google_cloud_run_service.self[each.key].name
+}
+
+//noinspection HILUnresolvedReference
+resource "google_cloud_run_domain_mapping" "default" {
```

Comment:
I get the reason for default or other naming but favour self. I think self here as well?